### PR TITLE
<Update with how to do> allow-com-object-registration-in-windows-defender-application-…

### PR DESCRIPTION
**Pull request by TJ2215**
…control-policy.md

### CODE REVIEW

File: windows/security/threat-protection/windows-defender-application-control/allow-com-object-registration-in-windows-defender-application-control-policy.md [hunk 1]
```diff
@@ -92,4 +92,55 @@ Example 3: Allows a specific COM object to register in PowerShell
   </Value>
 </Setting>
 ```
+### How to configure the settings for the CLSIDs
+For example, you get an error in the Event Viewer (Application and Service Logs > Microsoft > Windows > AppLocker > MSI and Script) like below:
```
Comment (illfated) [inline windows/security/threat-protection/windows-defender-application-control/allow-com-object-registration-in-windows-defender-application-control-policy.md:96]: I don't know enough about this document to say if this addition will become accepted (I am not a Microsoft employee), but in the meanwhile, here are some improvement suggestions:
```suggestion
### How to configure settings for the CLSIDs

Given the following example of an error in the Event Viewer (Application and Service Logs > Microsoft > Windows > AppLocker > MSI and Script):
```

File: windows/security/threat-protection/windows-defender-application-control/allow-com-object-registration-in-windows-defender-application-control-policy.md [hunk 2]
```diff
@@ -92,4 +92,55 @@ Example 3: Allows a specific COM object to register in PowerShell
...
+Date:          11/11/2020 1:18:11 PM
+Event ID:      8036
+Task Category: None
+Level:         Error
+Keywords:      
+User:          S-1-5-21-3340858017-3068726007-3466559902-3647
+Computer:     contoso.com
+Description:
+{f8d253d9-89a4-4daa-87b6-1168369f0b21} was prevented from running due to Config CI policy.
+Event Xml:
```
Comment (illfated) [inline windows/security/threat-protection/windows-defender-application-control/allow-com-object-registration-in-windows-defender-application-control-policy.md:109]: Spacing is important. Removed redundant blanks, adjusted indentation, capitalization for XML (eXtensible Markup Language).
```suggestion
Keywords:
User:          S-1-5-21-3340858017-3068726007-3466559902-3647
Computer:      contoso.com
Description:
{f8d253d9-89a4-4daa-87b6-1168369f0b21} was prevented from running due to Config CI policy.

Event XML:

```

File: windows/security/threat-protection/windows-defender-application-control/allow-com-object-registration-in-windows-defender-application-control-policy.md [hunk 3]
```diff
@@ -92,4 +92,55 @@ Example 3: Allows a specific COM object to register in PowerShell
...
+    <Execution ProcessID="21060" ThreadID="23324" />
+    <Channel>Microsoft-Windows-AppLocker/MSI and Script</Channel>
+    <Computer>contoso.com</Computer>
+    <Security UserID="S-1-5-21-3340858017-3068726007-3466559902-3647" />
+  </System>
+  <EventData>
+    <Data Name="IsApproved">false</Data>
+    <Data Name="CLSID">{f8d253d9-89a4-4daa-87b6-1168369f0b21}</Data>
+  </EventData>
+</Event>
```
Comment (illfated) [inline windows/security/threat-protection/windows-defender-application-control/allow-com-object-registration-in-windows-defender-application-control-policy.md:131]:
```suggestion
```XML
<Event xmlns="http://schemas.microsoft.com/win/2004/08/events/event">
  <System>
    <Provider Name="Microsoft-Windows-AppLocker" Guid="{cbda4dbf-8d5d-4f69-9578-be14aa540d22}" />
    <EventID>8036</EventID>
    <Version>0</Version>
    <Level>2</Level>
    <Task>0</Task>
    <Opcode>0</Opcode>
    <Keywords>0x4000000000000000</Keywords>
    <TimeCreated SystemTime="2020-11-11T19:18:11.4029179Z" />
    <EventRecordID>819347</EventRecordID>
    <Correlation ActivityID="{61e3e871-adb0-0047-c9cc-e761b0add601}" />
    <Execution ProcessID="21060" ThreadID="23324" />
    <Channel>Microsoft-Windows-AppLocker/MSI and Script</Channel>
    <Computer>contoso.com</Computer>
    <Security UserID="S-1-5-21-3340858017-3068726007-3466559902-3647" />
  </System>
  <EventData>
    <Data Name="IsApproved">false</Data>
    <Data Name="CLSID">{f8d253d9-89a4-4daa-87b6-1168369f0b21}</Data>
  </EventData>
</Event>
```
I have tried adding a MarkDown fenced code block with syntax highlighting for XML, but I am not sure if I have been successful in adding the ending fence (3 back ticks, **```**)

File: windows/security/threat-protection/windows-defender-application-control/allow-com-object-registration-in-windows-defender-application-control-policy.md [hunk 4]
```diff
@@ -92,4 +92,55 @@ Example 3: Allows a specific COM object to register in PowerShell
...
+
+To add this CLSID to the existing policy, follow the steps below,
+1. Open the Powershell ISE with administrative priviledge.
+2. Now from the admin powershell ISE, type this command and run it. Considering the name of the policy is WDAC_policy.xml .                                                                                                                                    
+PS C:\WINDOWS\system32> Set-CIPolicySetting -FilePath <path to policy xml>\WDAC_policy.xml -Key 8856f961-340a-11d0-a96b-00c04fd705a2 -Provider WSH -Value True -ValueName EnterpriseDefinedClsId -ValueType Boolean
+
+Once the command is run, you will find that the following section is added to the policy xml.
```
Comment (illfated) [inline windows/security/threat-protection/windows-defender-application-control/allow-com-object-registration-in-windows-defender-application-control-policy.md:138]:
~~~suggestion
To add this CLSID to the existing policy, use the following steps:

1. Open PowerShell ISE with Administrative privileges.
2. From the admin PowerShell ISE, copy and edit this command, then run it. Consider the policy name to be `WDAC_policy.xml`.

```PowerShell
PS C:\WINDOWS\system32> Set-CIPolicySetting -FilePath <path to policy xml>\WDAC_policy.xml -Key 8856f961-340a-11d0-a96b-00c04fd705a2 -Provider WSH -Value True -ValueName EnterpriseDefinedClsId -ValueType Boolean
```

Once the command has been run, you will find that the following section is added to the policy XML.
~~~

File: windows/security/threat-protection/windows-defender-application-control/allow-com-object-registration-in-windows-defender-application-control-policy.md [hunk 5]
```diff
@@ -92,4 +92,55 @@ Example 3: Allows a specific COM object to register in PowerShell
...
+
+  <Settings>
+    <Setting Provider="WSH" Key="8856f961-340a-11d0-a96b-00c04fd705a2" ValueName="EnterpriseDefinedClsId">
+      <Value>
+        <Boolean>true</Boolean>
+      </Value>
+    </Setting>
```
Comment (illfated) [inline windows/security/threat-protection/windows-defender-application-control/allow-com-object-registration-in-windows-defender-application-control-policy.md:145]:
~~~suggestion
```XML
  <Settings>
    <Setting Provider="WSH" Key="8856f961-340a-11d0-a96b-00c04fd705a2" ValueName="EnterpriseDefinedClsId">
      <Value>
        <Boolean>true</Boolean>
      </Value>
    </Setting>
```
~~~

**Comment by illfated** [comment]
I can only apologize for the fact that GitHub is unable to highlight the individual changes for slightly big sections in suggestions. The other option would be to add the suggestions line by line, or max 2 lines at a time (which would result in quite a lot of commits). Feel free to tell me if you would prefer many commits to for easier review of the individual edits.

File: windows/security/threat-protection/windows-defender-application-control/allow-com-object-registration-in-windows-defender-application-control-policy.md [hunk 6]
```diff
@@ -92,4 +92,55 @@ Example 3: Allows a specific COM object to register in PowerShell
...
+    <Computer>contoso.com</Computer>
+    <Security UserID="S-1-5-21-3340858017-3068726007-3466559902-3647" />
+  </System>
+  <EventData>
+    <Data Name="IsApproved">false</Data>
+    <Data Name="CLSID">{f8d253d9-89a4-4daa-87b6-1168369f0b21}</Data>
+  </EventData>
+</Event>
+
+To add this CLSID to the existing policy, follow the steps below,
```
Comment (JohanFreelancer9) [inline windows/security/threat-protection/windows-defender-application-control/allow-com-object-registration-in-windows-defender-application-control-policy.md:133]:
```suggestion
You can add this CLSID to the existing policy by following these steps:
```

**Review by JohanFreelancer9 — CHANGES_REQUESTED** [review]
Copy review - @illfated, your suggestions look good, I have just added a few more. I'll do a final edit after the technical review.

File: windows/security/threat-protection/windows-defender-application-control/allow-com-object-registration-in-windows-defender-application-control-policy.md (hunk 1 above)
Comment (JohanFreelancer9) [inline windows/security/threat-protection/windows-defender-application-control/allow-com-object-registration-in-windows-defender-application-control-policy.md:96]:
```suggestion
For example, should you get an error in the Event Viewer (**Application and Service Logs** > **Microsoft** > **Windows** > **AppLocker** > **MSI and Script**) such as the following:
```

**Comment by andreiztm** [comment]
Thanks everyone for the help so far with this!

**Comment by illfated** [comment]
@TJ2215 : Would you be able to commit the suggested changes, or apply your own changes as close as possible to our suggestions, so we can move this PR forward to being merge ready?

**Comment by TJ2215** [comment]
@illfated  sure.. let me make those changes

File: windows/security/threat-protection/windows-defender-application-control/allow-com-object-registration-in-windows-defender-application-control-policy.md [hunk 7]
```diff
@@ -92,4 +91,66 @@ Example 3: Allows a specific COM object to register in PowerShell
...
+    <Computer>contoso.com</Computer>
+    <Security UserID="S-1-5-21-3340858017-3068726007-3466559902-3647" />
+  </System>
+  <EventData>
+    <Data Name="IsApproved">false</Data>
+    <Data Name="CLSID">{f8d253d9-89a4-4daa-87b6-1168369f0b21}</Data>
+  </EventData>
+</Event>
+```
+
```
Comment (illfated) [inline windows/security/threat-protection/windows-defender-application-control/allow-com-object-registration-in-windows-defender-application-control-policy.md:136]: Cosmetic/codestyle: we can remove one blank line here to keep the spacing consistent (although not terribly important).
```suggestion
```

File: windows/security/threat-protection/windows-defender-application-control/allow-com-object-registration-in-windows-defender-application-control-policy.md [hunk 8]
```diff
@@ -92,4 +91,66 @@ Example 3: Allows a specific COM object to register in PowerShell
   </Value>
 </Setting>
 ```
+### How to configure settings for the CLSIDs
+
+Given the following example of an error in the Event Viewer (Application and Service Logs > Microsoft > Windows > AppLocker > MSI and Script):
```
Comment (illfated) [inline windows/security/threat-protection/windows-defender-application-control/allow-com-object-registration-in-windows-defender-application-control-policy.md:96]: I also suggest applying the following part of @JohanFreelancer9's earlier suggestion:
```suggestion
Given the following example of an error in the Event Viewer (**Application and Service Logs** > **Microsoft** > **Windows** > **AppLocker** > **MSI and Script**):
```

**Review by JohanFreelancer9 — APPROVED** [review]

**Review by illfated — APPROVED** [review]
Seems ready for final review & merge (unless anyone in the MS Docs management should have any arguments against it).

**Comment by TJ2215** [comment]
thanks everyone for their contribution. I believe this is ready for the final review by @isbrahm and team.

**Pull request merged**